src/compiler/scala/tools/nsc/typechecker/RefChecks.scala

```diff
@@ -1273,6 +1273,9 @@ abstract class RefChecks extends Transform {
         // The getOrElse part should never happen, it's just here as a backstop.
         reporter.error(pos, sym.compileTimeOnlyMessage getOrElse defaultMsg)
       }
+      // More blacklisted nasties.
+      if (sym == Object_asInstanceOf || sym == Object_isInstanceOf)
+        reporter.error(pos, s"Bad reference to internal symbol: $sym")
```

Comment:
i'm wondering if, instead, it would be cleaner to avoid entering those methods in scope? https://github.com/scala/scala/compare/2.13.x...lrytz:h%C3%A4ck?expand=1 (we'd need to do it nicer)

Reply:
Also my first thought, but at midnight, this was quickier. I checked that completion no longer shows them. There are lots of dollar things they shouldn't mess with.